test/lit/passes/dce-gc.wast

```diff
@@ -0,0 +1,104 @@
+;; NOTE: Assertions have been generated by update_lit_checks.py and should not be edited.
+;; RUN: wasm-opt %s --dce -all -S -o - | filecheck %s
+
+(module
+  ;; CHECK:      (func $param1 (param $ignore nullfuncref) (param $no (ref nofunc))
+  ;; CHECK-NEXT:  (drop
+  ;; CHECK-NEXT:   (local.get $ignore)
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT:  (block
+  ;; CHECK-NEXT:   (drop
+  ;; CHECK-NEXT:    (local.get $no)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:   (unreachable)
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $param1 (param $ignore (ref null nofunc)) (param $no (ref nofunc))
```

Comment:
We could be even more aggressive (and save some compile time!) by replacing the entire body of any function that takes an impossible argument with `unreachable`.

Reply:
Makes sense.

I'm rethinking this though, given we should use `unreachable` in the long term. Maybe this isn't worth the work to scan the entire function another time. It might make more sense as a peephole optimization actually.

Reply:
Yeah, I think this general optimization could make sense as a peephole optimization as well. The specific optimization for the impossible parameters still makes sense as its own pass, I think.